test/http-response-message.spec.js

```diff
@@ -1,155 +1,203 @@
 
       expect(httpResponse.requestMessage).toBe(requestMessage);
-      expect(httpResponse.statusCode).toBe(xhr.status);
       expect(httpResponse.response).toBe(xhr.response);
+      expect(httpResponse.statusCode).toBe(xhr.status);
       expect(httpResponse.statusText).toBe(xhr.statusText);
-      expect(httpResponse.responseType).toBe(responseType);
       expect(httpResponse.reviver).toBe(reviver);
       expect(httpResponse.headers).toEqual(jasmine.any(Headers));
     });
 
-    it("will call the Headers.parse() if xhr.getAllResponseHeaders() is defined", () => {
-      let parseSpy = spyOn(Headers, 'parse');
-      let xhrMock = jasmine.createSpyObj("xhr", ["getAllResponseHeaders"]);
-      xhrMock.getAllResponseHeaders.and.returnValue("");
-
-      new HttpResponseMessage({}, xhrMock);
-      expect(xhrMock.getAllResponseHeaders).toHaveBeenCalled();
-      expect(parseSpy).toHaveBeenCalledWith("");
-    });
-
-    it("will set responseType to an aliased version of the mimetype detected in the Content-Type header, will also store the original mimetype as mimeType", () => {
-      runContentTypeExpectations([
-        {contentType:"text/yml",mimeType:"text/yml",type:"yml"},
-        {contentType:"text/xml",mimeType:"text/xml",type:"xml"},
-        {contentType:"text/markdown",mimeType:"text/markdown",type:"md"}
-      ]);
+    it('parses headers', () => {
+      var parseSpy = spyOn(Headers, 'parse').and.callThrough();
+      new HttpResponseMessage({}, mockXhr());
+      expect(parseSpy).toHaveBeenCalled();
     });
 
-    it("will split the Content-Type using the ; delimeter and use the first part as the mimeType", () => {
+    it('sets responseType to an aliased version of the mimetype in the Content-Type header and the mimeType to the original mimetype', () => {
       runContentTypeExpectations([
-        {contentType:"text/html; charset=utf-8",mimeType:"text/html",type:"html"},
-        {contentType:"application/atom+xml; type=feed",mimeType:"application/atom+xml",type:"atom"},
-        {contentType:"application/json;   odata=verbose",mimeType:"application/json",type:"json"}
+        { contentType: 'text/yml',      mimeType: 'text/yml',      type: 'yml' },
+        { contentType: 'text/xml',      mimeType: 'text/xml',      type: 'xml' },
+        { contentType: 'text/markdown', mimeType: 'text/markdown', type: 'md'  }
       ]);
     });
 
-    it("will use the mimeType as the responseType if there is no alias for it in the mimeTypes map", () => {
+    it('splits the Content-Type header using the ; delimeter and sets mimeType to the first part', () => {
       runContentTypeExpectations([
-        {contentType:"text/foo",mimeType:"text/foo",type:"text/foo"},
-        {contentType:"text/foo; charset=utf-8",mimeType:"text/foo",type:"text/foo"}
+        { contentType:'text/html; charset=utf-8',          mimeType:'text/html',            type:'html' },
```

Comment:
It'd be nice to format consistently. This object is missing spaces after colons, where spaces are used above.